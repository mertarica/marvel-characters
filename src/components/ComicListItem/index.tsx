import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { IComic } from "../../types/comic.type";

type Props = {
  comic: IComic;
};

const ComicListItem = ({ comic }: Props) => {
  return (
    <Row className="c-comic-list-item">
      <Col lg={3}>
        <img
          src={`${comic.thumbnail?.path}.${comic.thumbnail?.extension}`}
          alt={`${comic.title} cover`}
          className="w-100"
        />
      </Col>
      <Col lg={9}>
        <h3 className="c-comic-list-item__header">{comic.title}</h3>
        <p className="c-comic-list-item__description">{comic.description}</p>
      </Col>
    </Row>
  );
};

export default ComicListItem;
