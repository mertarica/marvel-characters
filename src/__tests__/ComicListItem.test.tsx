import renderer from "react-test-renderer";
import ComicListItem from "../components/ComicListItem";
import fakeComic from "../__mock__/comic";

describe("<ComicListItem />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ComicListItem comic={fakeComic} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
