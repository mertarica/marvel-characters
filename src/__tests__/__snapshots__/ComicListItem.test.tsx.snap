// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicListItem /> renders correctly 1`] = `
<div
  className="c-comic-list-item row"
>
  <div
    className="col-lg-3"
  >
    <img
      alt="Fake Hero cover"
      className="w-100"
      src="."
    />
  </div>
  <div
    className="col-lg-9"
  >
    <h3
      className="c-comic-list-item__header"
    >
      Fake Hero
    </h3>
    <p
      className="c-comic-list-item__description"
    >
      Fake hero description
    </p>
  </div>
</div>
`;
